src/connector/src/sink/mod.rs

```diff
@@ -66,6 +82,29 @@ pub trait Sink {
     async fn abort(&mut self) -> Result<()>;
 }
 
+pub struct SinkWriterInfo {
+    // TODO: add fields
+}
+
+#[async_trait]
+pub trait SinkCoordinator {
+    async fn init(&mut self, writer_info: Vec<SinkWriterInfo>) -> Result<()>;
+    async fn commit(&mut self, epoch: u64, metadata: Vec<Bytes>) -> Result<()>;
```

Comment:
Why is `metadata` an array of bytes instead of a more meaningful struct here?

Reply:
The array is a collection of all metadata collected from each parallelism of the sink writer, and the metadata of each parallelism is Bytes because the metadata is passed through RPC, and different sink have different metadata definition. Therefore, here the Bytes is a serialized metadata. The sink writer will serialize its own metadata and the coordinator should deserialize the metadata.

Reply:
then it is better to add some comments here. quite confusing
IIRC, only datalake sink will return the metadata and it should be filenames?